### DISCUSSION TITLE
How to serve index.html file

### DISCUSSION THREAD

**Discussion by FromCppToCarbon** [oven-sh/bun#1392] [General]
May i know how to serve index.html file in bun ?  with advance thanks for the answer

**Comment by Jarred-Sumner** [comment]
Here is an example:

```ts
Bun.serve({
  fetch(req) {
   const url = new URL(req.url);
    if (url.pathname.endsWith("/") || url.pathname.endsWith("/index.html"))
       return new Response(Bun.file(import.meta.dir + "/index.html"));
    
     // all other routes
     return new Response("Hello!");
  }
});
```

**Reply by enbeec** [reply]
Can this be done with the file loader?

**Comment by 1jackdev** [comment]
is this what you are looking for?
```
Bun.serve({
  fetch(req: Request): Response | Promise<Response> {
    return new Response(Bun.file("./index.html"));
  },

  port: process.env.PORT || 3000,
});

```

**Comment by alber70g** [comment]
How to replicate `bun ./**/*.html` programmatically? I'm looking to host for a small local project on `0.0.0.0` instead of just `localhost` so I can access it from anywhere on the local network.

**Reply by alber70g** [reply]
Or `bun create ./Component.tsx` for that matter
